posts/2022/05/the-evolution-of-scipy-cli.md

```diff
@@ -0,0 +1,157 @@
+        },
+    ],
+
+    "do.py test": [
+        {
+            "name": "Options",
+            "options": ["--help", "--verbose", "--parallel", "--coverage"],
+        },
+        {
+            "name": "Options: test selection",
+            "options": ["--submodule", "--tests", "--mode"],
+        },
+    ],
+}
+```
+
+#### 🎥 Devloper CLI in action
+
+<script id="asciicast-U9l9VvklvEjXdEi1xYS4A7u5M" src="https://asciinema.org/a/U9l9VvklvEjXdEi1xYS4A7u5M.js" async></script>
+
+#### Current list of implemented tasks
+* `Build & testing tasks`
+     - build (build & install package on path)
+     - test (Run tests along with options to run tests for a given submodule)
+* `Static checker tasks`
+     - pep8 ( Perform pep8 check with flake8)
+     - mypy ( Run mypy on the codebase)
+* `Environments`
+     - shell (Start Unix shell with PYTHONPATH set)
+     - python (Start a Python shell with PYTHONPATH set)
+     - ipython (Start IPython shell with PYTHONPATH set )
+* `Documentation tasks`
+     - doc (Build documentation)
+     - refguide-check  (Run refguide check)
+ 
+* `Release tasks`
+    -  notes   (Release notes and log generation)               
+    -  authors  (Task to generate list the authors who contributed within a given revision interval)
+* `Benchmarking tasks`
+    - bench & bench compare
+
+#### 👏🏽 Great Collaboration and teamwork
+
+From an idea to developing a successful POC, the journey was a great learning opportunity for me. Planning, coordination, teamwork and clear communication played a very important role. Huge thanks to **Ralf Gommers** and **Eduardo Naufel Schettino** for the amazing collaboration and support. 
+
+As a newcomer to the `SciPy codebase`, it was a steep learning curve. I asked a ton of questions, at times drifted to the ocean searching for answers. Learning something completely new be overwhelming, different emotions brush past. Thanks to Ralf and **Pamphile Roy** for addressing my questions. The learnings and achievements will stay with me for a long-long time.    
+
+#### 😇 The possibilities next steps
+
+The experimental CLI is available under `scipy/do.py` for the wider community to test and provide us with valuable feedback.
+Some of the handy commands to quickly try out the CLI -
+- Enabling the GUI: `python do.py`
+- Listing the args/options for a task: `python do.py <task_name> --help`
+
+**Outcomes**
+
+1. 📜 Self documentation and hierarchical help option
+2. 🧭 Easy to navigate and intuitive interface 
+3. ⏩ Clear and concise examples to get started quickly
+4. ⏱ Reduction in the time spent navigating documentation
+
+With a great start comes possibilities. In the coming weeks, the CLI will become more mature and stable. After we receive wider usage and acceptance from the community, support for `dev.py` and `runtests.py` will be paused and `do.py` will be renamed to `dev.py`. The user documentation for the CLI components and usage will be made available for clear and concise understanding. 
```

Comment:
Other next steps: `pydevtool` work that Eduardo is doing to make things more reusable. Adding support for other features, like running GitHub CI jobs via Act.